src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -78,6 +78,11 @@ public Conversion ClassifyImplicitConversionFromExpression(BoundExpression sourc
             Debug.Assert(sourceExpression != null);
             Debug.Assert((object)destination != null);
 
+            if (sourceExpression.Kind == BoundKind.ExpressionWithNullability)
+            {
+                sourceExpression = ((BoundExpressionWithNullability)sourceExpression).Expression;
+            }
```

Comment:
I think you caught a problem. I'll try to make a scenario. 
We probably do this unwrapping only for `ClassifyImplicitBuiltInConversionFromExpression` rather than for all cases in this method (for instance, the `GetImplicitUserDefinedConversion` probably should operate on the flow-analyzed type).